### PR TITLE
Update model tuning for regression tasks and add new models

### PR DESCRIPTION
**Pull request by lxywizard**
Some minor updates:
- Since regression tasks use RMSE instead of accuracy, the criterion of determining the `best_model_score` should be inverted as well as the result passed to the advisor.
- Add new `TABULAR_CLASSIFICATION` models for testing OBOE.

### CODE REVIEW

File: rafiki/model/dev.py [hunk 1]
```diff
@@ -104,18 +108,28 @@ def tune_model(py_model_class: Type[BaseModel], train_dataset_path: str, val_dat
...
+                    else (result.score > best_model_score)
+            if is_best_model:
+                inform_user('Best saved model so far! Beats previous best of score {}!'.format(best_model_score))
+                best_store_params_id = store_params_id
+                best_proposal = proposal
+                best_model_score = result.score
+                best_trial_no = trial_no
+
+                # Need to minimize RMSE
+                if train_args['task_type'] == 'regression':
```
Comment (nginyc) [inline rafiki/model/dev.py]: Editing the tuning logic here is simply edits it for *local development* by model developers, but doesn't update the logic for model training workers that are actually deployed on Rafiki. Moreover, I believe we should do the inversion of the RMSE *within the `evaluate` method* of the model to avoid special-casing like this - and it conforms to the [method's description](https://nginyc.github.io/rafiki/docs/0.2.0/src/python/rafiki.model.html#rafiki.model.BaseModel.evaluate)

**Comment by nginyc** [comment]
Also need to merge latest changes from `dev`. The latest `dev` docs are here: https://nginyc.github.io/rafiki/docs/0.2.0/index.html

File: rafiki/model/dev.py (hunk 1 above)
Comment (lxywizard) [reply rafiki/model/dev.py]: I agree with your opinion on inverting the RMSE for the consistency of the `evaluate` method.

**Comment by lxywizard** [comment]
The new models are added in #149 and the model tuning for regression tasks will be dealt with later.
